plugins/ilert/src/components/ServicesPage/ServicesTable.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import React from 'react';
+import { TableState } from '../../api';
+import { Service } from '../../types';
+import { StatusChip } from './StatusChip';
+
+import { Table, TableColumn } from '@backstage/core-components';
+import { ServiceActionsMenu } from '../Service/ServiceActionsMenu';
+import { ServiceLink } from '../Service/ServiceLink';
+
+const useStyles = makeStyles(theme => ({
+  empty: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    justifyContent: 'center',
+  },
+}));
+
+export const ServicesTable = ({
+  services,
+  tableState,
+  isLoading,
+  onChangePage,
+  onChangeRowsPerPage,
+  compact,
+}: {
+  services: Service[];
+  tableState: TableState;
+  isLoading: boolean;
+  setIsLoading: (isLoading: boolean) => void;
+  onChangePage: (page: number) => void;
+  onChangeRowsPerPage: (pageSize: number) => void;
+  compact?: boolean;
+}) => {
+  const classes = useStyles();
+
+  // const xsColumnStyle = {
+  //   width: '5%',
+  //   maxWidth: '5%',
+  // };
+  const smColumnStyle = {
+    width: '10%',
+    maxWidth: '10%',
+  };
+  // const mdColumnStyle = {
+  //   width: '15%',
+  //   maxWidth: '15%',
+  // };
+  // const lgColumnStyle = {
+  //   width: '20%',
+  //   maxWidth: '20%',
+  // };
+  const xlColumnStyle = {
+    width: '30%',
+    maxWidth: '30%',
+  };
+
+  const idColumn: TableColumn = {
```

Comment:
A tip for not having to do the casting to Service is to use `TableColumn<Service>` for all of these. Then the `render` function will have `rowData` of the right type.

Reply:
Definitely smarter and cleaner code, thanks.